### PR TITLE
Add a tool that allows us to quickly update specification URLs to a s…

### PR DESCRIPTION
**Pull request by jsuereth**
…pecific version.

This should make it a lot easier for us to direct link against the specification and rapidly update links/check them over time.

### CODE REVIEW

**Comment by Oberon00** [comment]
Note that there is an issue for that at the semantic convention generator: https://github.com/open-telemetry/build-tools/issues/20 But since this is now needed also outside the generated markdown, a separate tool might be better anyways. Do consider adding it to the build-tools repository though (or should the semconv generator move here?)

**Comment by jsuereth** [comment]
@Oberon00 very good questions.  I like to start small and evolve over time, so I think since the script is short and directly tied to semconv maintenance, we can leave it here.

If we discover limitations that necessitate moving it into build-tools I'm more than happy to move the tool over there.

Regarding the evolution of build-tools and this repository, it COULD be we merge them together over time.  Let's see how "momentum" goes as we migrate to this new repository and if we find huge friction keeping it separate then we can push for merging.  

It's a great call out, let's have everyone keep the idea on our radar.

File: internal/tools/update_specification_version.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Example usage:
+#
+# ./internal/tools/update_specification_version.sh
+
+
+# Set this to the version number you want to CHANGE in URLs in the repository.
+PREVIOUS_SPECIFICATION_VERSION="main"
+# Set this tot he version number you want to KEEP in URLs in the repository.
```
Comment (lmolkova) [inline internal/tools/update_specification_version.sh:10]:
```suggestion
# Set this to the version number you want to KEEP in URLs in the repository.
```

File: internal/tools/update_specification_version.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+LATEST_SPECIFICATION_VERSION="1.20.0"
+# This the the specific pattern we look for when replacing URLs
```
Comment (lmolkova) [inline internal/tools/update_specification_version.sh:12]:
```suggestion
# This the specific pattern we look for when replacing URLs
```

**Review by lmolkova — APPROVED** [review]

**Comment by Oberon00** [comment]
I like the `find` solution, however, I wonder if we should add a check against adding files with spaces to the repository instead of trying to support spaces here. You probably don't want paths with spaces anyways, and somewhere you most likely will run into a bug anyways.

**Review by Oberon00 — APPROVED** [review]

**Review by pyohannes — APPROVED** [review]

**Comment by jsuereth** [comment]
Thanks for the improvements!

Regarding spaces in files, remember that's *common* in windows that your home directory may even have spaces.  When using WSL / cygwin, it's really important to handle regardless.

**Pull request merged**